astropy/units/equivalencies.py

```diff
@@ -561,6 +564,344 @@ def mass_energy():
+
+        Parameters
+        ----------
+        source_quantity: float
+            source quantity value
+        target_unit: astropy.units.Quantity, astropy.units.Unit
+            list of units
+        """
+        source_quantity_decomposed = source_quantity.decompose()
+        target_unit_decomposed = target_unit.decompose()
+
+        bases = source_quantity_decomposed.unit.bases
+        powers = source_quantity_decomposed.unit.powers
+
+        try:
+            target_unit_powers = target_unit_decomposed.powers
+            target_unit_base = target_unit_decomposed.bases[0]
+        except AttributeError:
+            target_unit_powers = target_unit_decomposed.unit.powers
+            target_unit_base = target_unit_decomposed.unit.bases[0]
+
+        # sanity check
+        assert all(hasattr(self, b.__str__()) for b in bases)
+        assert sum(powers) == sum(target_unit_powers)
+
+        res = source_quantity_decomposed.value
+        for b, p in zip(bases, powers):
+            res *= self._homogeneous_conversion(
+                1.0 * b**p, target_unit_base**p
+            )
+        return res
+
+
+def geometrized(physical_type):
+    """
+    Convert quantities to geometric units.
+
+    Parameters
+    ----------
+    physical_type: str or `~astropy.units.PhysicalType`
+        Physical type to convert.
+        Argument to :meth:`astropy.units.get_physical_type`
+
+    Examples
+    --------
+    >>> from astropy import constants as c, units as u
+    >>> mass = 2 * u.M_sun
+    >>> mass.to('km', equivalencies=u.geometrized('mass'))
+    <Quantity 2.95325008 km>
+    >>> time = 1 * u.microsecond
+    >>> time.to('solMass', equivalencies=u.geometrized('time'))
+    <Quantity 0.20302545 solMass>
+    >>> time.to('kg', equivalencies=u.geometrized('time'))
+    <Quantity 4.03697802e+29 kg>
+    """
+    physical_type = physical.get_physical_type(physical_type)
+    mass_length_time_physical_types = [
+        physical.length,
+        physical.mass,
+        physical.time,
+    ]
+
+    volume_physical_types = [
+        physical.volume
+    ]
+
+    area_physical_types = [
+        physical.area,
+    ]
+
+    temperature_physical_types = [
+        physical.temperature,
+    ]
+
+    velocity_physical_types = [
+        physical.velocity,
+    ]
+
+    mass_density_physical_type = [
+        physical.mass_density
+    ]
+
+    allowed_physical_types = mass_length_time_physical_types
+    allowed_physical_types += volume_physical_types
+    allowed_physical_types += area_physical_types
+    allowed_physical_types += temperature_physical_types
+    allowed_physical_types += velocity_physical_types
+    allowed_physical_types += mass_density_physical_type
+
+    assert physical_type in allowed_physical_types, (
+        "Allowed values : "
+        f"{[v._physical_type for v in allowed_physical_types]}"
+    )
+
+    conversions = GeometricUnitConversionFactors()
+
+    if physical_type in velocity_physical_types:
```

Comment:
This linked if-elseif statement could I think more logically be done with a `dict` that is keyed by physical_type (they are hashable, so that should work), i.e., 
```
return geometrized_equivalencies[physical_type]
```